modules/core/src/lib/effect-manager.js

```diff
@@ -63,4 +67,46 @@ export default class EffectManager {
     }
     this.needApplyDefaultLighting = !hasEffect;
   }
+
+  _checkShadowEffect() {
+    let shouldApplyShadow = false;
+    for (const effect of this.effects) {
+      if (effect instanceof LightingEffect && effect.directionalLights.length > 0) {
+        shouldApplyShadow = true;
+        break;
+      }
+    }
+    this.needApplyDefaultShadow = shouldApplyShadow;
+  }
+
+  _applyShadowEffect(effects) {
+    const lights = [];
+    for (const effect of this.effects) {
+      if (effect instanceof LightingEffect) {
+        for (const light of effect.directionalLights) {
+          // eslint-disable-next-line max-depth
+          if (light.castShadow) {
+            lights.push(light);
+          }
+        }
+      }
+    }
+    effects.push(new ShadowEffect({lights}));
+  }
+
+  _applyLightingEffect() {
+    this.internalEffects.push(this.defaultLightingEffect);
+  }
+
+  _createInternalEffects() {
```

Comment:
It seems weird to me that the effects manager is explicitly checking for each effect. Will this have to be done for all effects we add?

Reply:
no, not all the effect, only the effects user doesn't need to add and we need have a default one. Currently only the lighting effect.